src/components/Login/Login.tsx

```diff
@@ -0,0 +1,154 @@
+
+const { Text } = Typography;
+
+function Login() {
+  const [username, setUsernam] = useState('');
+  const [pass, setPass] = useState('');
+  const [logReg, setLogReg] = useState({
+    title: 'Log-in',
+    titleTwo: 'Register for first',
+    btnValue: 'Log-in',
+    status: true,
+  });
+
+  const authCtx = useContext(AuthContext);
+  const history = useNavigate();
+
+  const { Meta } = Card;
+
+  const userChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUsernam(e.target.value);
+  };
+  const passChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPass(e.target.value);
+  };
+  const submitHandler = () => {
+    setUsernam('');
+    setPass('');
+    if (logReg.status === false) {
+      fetch(
+        'https://identitytoolkit.googleapis.com/v1/accounts:signUp?key=AIzaSyBRegmu6l6G00JTkDtLcmhD97rD2NBg2cU',
+        {
+          method: 'POST',
+          body: JSON.stringify({
+            email: username,
+            password: pass,
+            returnSecueToken: true,
+          }),
+          headers: {
+            'Content-type': 'application/json',
+          },
+        }
+      )
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          } else {
+            return res.json().then(() => {
+              const errorMessage = 'Authentication failed';
+              throw new Error(errorMessage);
+            });
+          }
+        })
+        .then((data) => {
+          console.log(data);
+          console.log(data.idToken);
+          alert('Successfully signed up');
+        })
+        .catch((err) => {
+          alert(err.message);
+        });
+    } else {
+      fetch(
+        'https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyBRegmu6l6G00JTkDtLcmhD97rD2NBg2cU',
+        {
+          method: 'POST',
+          body: JSON.stringify({
+            email: username,
+            password: pass,
+            returnSecueToken: true,
+          }),
+          headers: {
+            'Content-type': 'application/json',
+          },
+        }
+      )
+        .then((res) => {
+          if (res.ok) {
+            return res.json();
+          } else {
+            return res.json().then(() => {
+              const errorMessage = 'Authentication failed';
+              throw new Error(errorMessage);
+            });
+          }
+        })
+        .then((data) => {
+          console.log(data);
+          console.log(data.idToken);
+          alert('Welcome ' + data.email);
+          authCtx.login(data.idToken);
+          console.log(authCtx);
+          history('/films');
+        })
+        .catch((err) => {
+          alert(err.message);
+        });
+    }
+  };
+  const changeHandler = () => {
+    setLogReg({
+      ...logReg,
+      title: 'Register',
+      titleTwo: 'Already registeered? So log-in',
+      btnValue: 'Register',
+      status: false,
+    });
+    if (logReg.status == false) {
```

Comment:
you have already private route I do not think something like this is needed but if there is a need you can explain here in the comment

Reply:
logReg.status is the status of user registration, this is checking is user registered or not. if this is false app provides us registration form , if true -  login form.